esphome/components/ld2410s/LD2410S.cpp

```diff
@@ -0,0 +1,947 @@
+      break;
+
+    case GATE_HOLD_THRESHOLD_WRITE_REPLY:
+      ESP_LOGI(TAG, "Trigger Hold written");
+      break;
+
+    case GATE_SNR_WRITE_REPLY:
+      ESP_LOGI(TAG, "Trigger SNR written");
+      break;
+
+    case OUTPUT_MODE_SWITCH_REPLY:
+      ESP_LOGI(TAG, "Minimal Output Mode switched");
+      break;
+
+    default:
+      ESP_LOGW(TAG, "< Unknown: %4x", command_word);
+      break;
+  }
+}
+CmdAckT LD2410S::parse_cms_frame_(uint8_t *buffer, size_t length) {
+  CmdAckT result;
+  size_t start = -1;
+  for (size_t i = 0; i < length; i++) {
+    if (memcmp(&buffer[i], &CMD_FRAME_HEADER, sizeof(CMD_FRAME_HEADER)) == 0) {
+      start = i;
+      break;
+    }
+  }
+  if (start == -1) {
+    ESP_LOGE(TAG, "Can't find cmd header");
+    result.result = false;
+    return result;
+  }
+  int data_length = this->two_byte_to_int_(buffer[start + 4], buffer[start + 5]);
+  result.length = data_length;
+  int command_word = this->two_byte_to_int_(buffer[start + 6], buffer[start + 7]);
+  result.command = command_word;
+  bool ack = buffer[start + 8] == 0x00 && buffer[start + 9] == 0x00;
+  result.result = ack;
+  for (size_t idx = 0; idx < result.length; idx++) {
+    memcpy(&result.data[idx], &buffer[idx + 10], sizeof(buffer[idx + 10]));
+  }
+  return result;
+}
+
+void LD2410S::process_ack_config_read_(uint8_t *data) {
+  this->max_dist_ = this->read_int_(data, 0, 4);
+  this->min_dist_ = this->read_int_(data, 4, 4);
+  this->delay_ = this->read_int_(data, 8, 4);
+  this->status_freq_ = this->read_int_(data, 12, 4);
+  this->dist_freq_ = this->read_int_(data, 16, 4);
+  this->resp_speed_ = this->read_int_(data, 20, 4);
+
+#ifdef USE_NUMBER
+  this->max_distance_number_->publish_state(static_cast<float>(this->max_dist_) * 0.7);
+  this->min_distance_number_->publish_state(static_cast<float>(this->min_dist_) * 0.7);
+  this->no_delay_number_->publish_state(this->delay_);
+  this->status_reporting_freq_number_->publish_state(static_cast<float>(this->status_freq_) / 10);
+  this->distance_reporting_freq_number_->publish_state(static_cast<float>(this->dist_freq_) / 10);
+#endif
+
+#ifdef USE_SELECT
+  this->response_speed_select_->publish_state(this->resp_speed_ == 5 ? RESPONSE_SPEED_NORMAL : RESPONSE_SPEED_FAST);
+#endif
+
+  ESP_LOGI(TAG,
+           "Config: max_dist=%d, min_dist=%d, delay=%d, status_resp_freq=%d, "
+           "dist_resp_freq=%d, resp_speed=%d",
+           this->max_dist_, this->min_dist_, this->delay_, this->status_freq_, this->dist_freq_, this->resp_speed_);
+}
+void LD2410S::process_ack_fw_read_(const uint8_t *data) {
+  int major_v = static_cast<int>(data[0]);
+  int minor_v = static_cast<int>(data[1]);
+  int patch_v = static_cast<int>(data[2]);
+  std::string version = "v" + std::to_string(major_v) + "." + std::to_string(minor_v) + "." + std::to_string(patch_v);
+
+  for (auto &listener : this->listeners_) {
+    listener->on_fw_version(version);
+  }
+
+  ESP_LOGI(TAG, "Firmware version: %s", version.c_str());
```

Comment:
This should be dumped in `dump_config()`, no need to log it here, too.
```suggestion
  ESP_LOGV(TAG, "Firmware version: %s", version.c_str());
```